pkg/lightweight/lightweight.go

```diff
@@ -184,21 +184,24 @@ func find(name string, origin string) map[string]*PartBag {
 
 			versionSort(versions)
 
-			// if oems were removable, there'd be know way of
+			typ := s.typ
+			inst := s.inst
+
+			// if oems were removable, there'd be no way of
 			// telling the kind of a removed origin-less package
 			if s.typ == pkg.TypeFramework && helpers.FileExists(filepath.Join(dirs.SnapOemDir, name)) {
```

Comment:
For reading continuity, can `s.typ` be changed to `typ`?

More importantly, I don't understand from this snippet how something that is of `TypeFramework` can ever be confused for `TypeOem`. And I wonder if this check is enough, is there a chance for `helpers.FileExists(filepath.Join(dirs.SnapFrameworkDir, name))` to return `true` as well?

Maybe a code comment is enough